File: E. LOGICAL TEST/Soal B.js
// nums is inputted array
// this code problem is same as leetcode problem, https://leetcode.com/problems/maximum-difference-between-increasing-elements/
// leetcode profile https://leetcode.com/WahyuWardana/

var maximumDifference = function (nums) {
	let max = 0;

	for (let i = 0, min = nums[0]; i < nums.length; i++) {
		min = Math.min(min, nums[i]);
		max = Math.max(max, nums[i] - min);
	}

	return max === 0 ? -1 : max;
};

File: E. LOGICAL TEST/SoalB.java
public class SoalB {
  public static int maximumDifference(int[] nums) {
    int maxDifference = 0;
    int min = nums[0];

    for (int num : nums) {
        min = Math.min(min, num);
        maxDifference = Math.max(maxDifference, num - min);
    }

    return maxDifference == 0 ? -1 : maxDifference;
  }

  public static void main(String[] args) {
      int[] nums = {7, 1, 5, 3, 2};
      int maxDifference = maximumDifference(nums);
      System.out.println(maxDifference);
  }
}
